ms18-applicatie/Controllers/team-a/CalendarController.cs

```diff
@@ -0,0 +1,184 @@
+            await requestUpdate.ExecuteAsync();
+            return new OkResult();
+        }
+
+        [HttpPost]
+        [Route("Event")]
+        public async Task<IActionResult> AddEvent(Calenders calenderName, CalenderEvent calendarEvent)
+        {
+            var googleEvent = calendarEvent.ToGoogleEvent();
+
+            EventsResource.InsertRequest request = _calendarService.Events.Insert(googleEvent, GetCalenderId(calenderName));
+            Event createdEvent = await request.ExecuteAsync();
+            return new OkResult();
+        }
+
+        [HttpGet]
+        [Route("welpen")]
+        public async Task<IActionResult> Welpen()
+        {
+            var events = await GetCalendar(GetCalenderId(Calenders.Welpen), false);
+            return new OkObjectResult(events);
+        }
+
+        [HttpGet]
+        [Route("matrozen")]
+        public async Task<IActionResult> Matrozen()
+        {
+            var events = await GetCalendar(GetCalenderId(Calenders.Matrozen), false);
+            return new OkObjectResult(events);
+        }
+
+        [HttpGet]
+        [Route("ZeeVerkenners")]
+        public async Task<IActionResult> ZeeVerkenners()
+        {
+            var events = await GetCalendar(GetCalenderId(Calenders.ZeeVerkenners), true);
+            return new OkObjectResult(events);
+        }
+
+        [HttpGet]
+        [Route("stam")]
+        public async Task<IActionResult> Stam()
+        {
+            var events = await GetCalendar(GetCalenderId(Calenders.Stam), false);
+            return new OkObjectResult(events);
+        }
+
+        [HttpGet]
+        [Route("global")]
+        public async Task<IActionResult> Global()
+        {
+            var events = await GetCalendar(GetCalenderId(Calenders.Stam), false);
+            return new OkObjectResult(events);
+        }
+
+        [HttpGet]
+        [Route("all")]
+        public async Task<IActionResult> All()
+        {
+            var events = new List<CalenderEvent>();
+            events.AddRange(await GetCalendar(GetCalenderId(Calenders.Matrozen)));
+            events.AddRange(await GetCalendar(GetCalenderId(Calenders.Welpen)));
+            events.AddRange(await GetCalendar(GetCalenderId(Calenders.ZeeVerkenners)));
+            events.AddRange(await GetCalendar(GetCalenderId(Calenders.Stam)));
+            events.AddRange(await GetCalendar(GetCalenderId(Calenders.Global)));
+            return new OkObjectResult(events);
+        }
+
+        private async Task<List<CalenderEvent>> GetCalendar(string calenderId, bool filterGlobal = true)
+        {
+            var calenderEvents = new List<CalenderEvent>();
+
+            var request = _calendarService.Events.List(calenderId);
+            var response = await request.ExecuteAsync();
+            var listItems = response.Items.ToList();
+
+            if (!filterGlobal)
+            {
+                request = _calendarService.Events.List(GetCalenderId(Calenders.Global));
```

Comment:
je overschrijft nu een eerder call die dus onnodig was misschien is een if/else hier beter dan een if